### PR TITLE
docs: clarify `redirects` on client-side navigation

### PR DESCRIPTION
**Pull request by balazsorban44**
Ref: #41344

## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Errors have a helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have a helpful link attached, see `contributing.md`

## Documentation / Examples

- [ ] Make sure the linting passes by running `pnpm lint`
- [ ] The "examples guidelines" are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md)

### CODE REVIEW

**Comment by luukdv** [comment]
Can you give an example of how the middleware should look to solve the [broken redirects issue](https://github.com/vercel/next.js/issues/40549)?

When using a basic `NextResponse.rewrite` in middleware, you're running into this issue: https://github.com/vercel/next.js/issues/40991

File: docs/api-reference/next.config.js/rewrites.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ module.exports = {
 }
 ```
 
-Rewrites are applied to client-side routing, a `<Link href="/about">` will have the rewrite applied in the above example.
+Rewrites are not applied to client-side routing (`Link`, `router.push`), unless [Middleware](/docs/advanced-features/middleware) is present.
```
Comment (ijjk) [inline docs/api-reference/next.config.js/rewrites.md:43]: `rewrites` are applied client-side regardless of middleware, `redirects` are not unless middleware is configured to match for that specific path

**Review by ijjk — CHANGES_REQUESTED** [review]
This change isn't accurate per https://github.com/vercel/next.js/pull/41362/files#r993802988

File: docs/api-reference/next.config.js/rewrites.md (hunk 1 above)
Comment (balazsorban44) [reply docs/api-reference/next.config.js/rewrites.md:43]: Looks like #40549 is a bug then. Will change the wording here :+1:

File: docs/api-reference/next.config.js/redirects.md [hunk 2]
```diff
@@ -47,12 +47,14 @@ module.exports = {
...
 
-- `basePath`: `false` or `undefined` - if false the basePath won't be included when matching, can be used for external rewrites only.
+- `basePath`: `false` or `undefined` - if false the basePath won't be included when matching, can be used for external redirects only.
 - `locale`: `false` or `undefined` - whether the locale should not be included when matching.
 - `has` is an array of [has objects](#header-cookie-and-query-matching) with the `type`, `key` and `value` properties.
 
 Redirects are checked before the filesystem which includes pages and `/public` files.
 
+Redirects are not applied to client-side routing (`Link`, `router.push`), unless [Middleware](/docs/advanced-features/middleware) is present.
```
Comment (ijjk) [inline docs/api-reference/next.config.js/redirects.md:56]:
```suggestion
Redirects are not applied to client-side routing (`Link`, `router.push`), unless [Middleware](/docs/advanced-features/middleware) is present and matches the path.
```

File: docs/api-reference/next.config.js/redirects.md [hunk 3]
```diff
@@ -47,12 +47,14 @@ module.exports = {
 
 > **Why does Next.js use 307 and 308?** Traditionally a 302 was used for a temporary redirect, and a 301 for a permanent redirect, but many browsers changed the request method of the redirect to `GET`, regardless of the original method. For example, if the browser made a request to `POST /v1/users` which returned status code `302` with location `/v2/users`, the subsequent request might be `GET /v2/users` instead of the expected `POST /v2/users`. Next.js uses the 307 temporary redirect, and 308 permanent redirect status codes to explicitly preserve the request method used.
 
-- `basePath`: `false` or `undefined` - if false the basePath won't be included when matching, can be used for external rewrites only.
+- `basePath`: `false` or `undefined` - if false the basePath won't be included when matching, can be used for external redirects only.
```
Comment (ijjk) [inline docs/api-reference/next.config.js/redirects.md:50]:
```suggestion
- `basePath`: `false` or `undefined` - if false the `basePath` won't be included when matching, can be used for external redirects only.
```

**Review by ijjk — APPROVED** [review]

**Pull request merged**